S3/Config.py

```diff
@@ -173,21 +173,29 @@ def role_config(self):
             error("IAM authentication not available -- missing module json")
             raise
         try:
-            conn = httplib.HTTPConnection(host='169.254.169.254', timeout = 2)
-            conn.request('GET', "/latest/meta-data/iam/security-credentials/")
-            resp = conn.getresponse()
-            files = resp.read()
-            if resp.status == 200 and len(files)>1:
-                conn.request('GET', "/latest/meta-data/iam/security-credentials/%s"%files)
-                resp=conn.getresponse()
-                if resp.status == 200:
-                    creds=json.load(resp)
-                    Config().update_option('access_key', creds['AccessKeyId'].encode('ascii'))
-                    Config().update_option('secret_key', creds['SecretAccessKey'].encode('ascii'))
-                    Config().update_option('access_token', creds['Token'].encode('ascii'))
+            container = os.environ.get("AWS_CONTAINER_CREDENTIALS_RELATIVE_URI", None)
+            if not container:
+                conn = httplib.HTTPConnection(host='169.254.169.254', timeout = 2)
+                conn.request('GET', "/latest/meta-data/iam/security-credentials/")
+                resp = conn.getresponse()
+                files = resp.read()
+                if resp.status == 200 and len(files)>1:
+                    conn.request('GET', "/latest/meta-data/iam/security-credentials/%s"%files)
+                    resp=conn.getresponse()
                 else:
                     raise IOError
             else:
+                conn = httplib.HTTPConnection(host='169.254.170.2', timeout = 2)
```

Comment:
@emilymdubois could you check this out against the payload you found in the curl response from 

```
169.254.170.2$AWS_CONTAINER_CREDENTIALS_RELATIVE_URI
```

I want to make sure that the code to parse the credentials from this file below is correct. That is, does the request to that URL return a JSON object like this?

```json
{
  "AccessKeyId": "xxx",
  "SecretAccessKey": "xxx",
  "Token": "xxx"
}
```